libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/gutenberg/GutenbergEditorFragment.java

```diff
@@ -795,6 +808,39 @@ public void onClick(DialogInterface dialog, int id) {
         dialog.show();
     }
 
+    private void showFeaturedImageConfirmationDialog(final int mediaId) {
+        if (mFeaturedImageId == 0) {
+            featuredImageSet(mediaId);
+        } else if (mFeaturedImageId != mediaId) {
+            Builder builder = new MaterialAlertDialogBuilder(getActivity());
+            builder.setTitle(R.string.replace_current_title);
+            builder.setMessage(R.string.replace_current_description);
+            builder.setPositiveButton(R.string.replace_current_confirmation,
+                    new DialogInterface.OnClickListener() {
+                        public void onClick(DialogInterface dialog, int id) {
+                            dialog.dismiss();
+                            featuredImageSet(mediaId);
+                        }
+                    });
+
+            builder.setNegativeButton(R.string.replace_current_cancel, new DialogInterface.OnClickListener() {
+                public void onClick(DialogInterface dialog, int id) {
+                    dialog.dismiss();
+                }
+            });
+
+            AlertDialog dialog = builder.create();
+            dialog.show();
+        }
+    }
+
+    private void featuredImageSet(int mediaId) {
+        mEditorFragmentListener.updateFeaturedImage(mediaId, true);
+        setFeaturedImageId(mediaId);
+        ToastUtils.showToast(getActivity(), R.string.featured_image_confirmation).show();
```

Comment:
👋 Siobhan, I'm assuming that the Toast here is mostly for debugging purposes and not part of the final design, right? If so, what do you think about adding a "TODO" comment here to help us remember to remove it before we finalize the feature?

Reply:
You're right, I'll add a "TODO" and this is a great lesson for me to be more diligent in doing similar in the future.